Can't compile in 6.9.6-arch1-1

I don't really think its a kernel version issue at all, make fails complaining about an implicit declaration:
```bash
make -C /lib/modules/6.9.6-arch1-1/build M=$PWD
make[1]: Entering directory '/usr/lib/modules/6.9.6-arch1-1/build'
  CC [M]  /home/skelly/nitara2/nitara2.o
/home/skelly/nitara2/nitara2.c:166:9: error: no previous prototype for ‘showmodule_read’ [-Werror=missing-prototypes]
  166 | ssize_t showmodule_read(struct file *unused_file, char *buffer, size_t len, loff_t *off)
      |         ^~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[3]: *** [scripts/Makefile.build:244: /home/skelly/nitara2/nitara2.o] Error 1
make[2]: *** [/usr/lib/modules/6.9.6-arch1-1/build/Makefile:1918: /home/skelly/nitara2] Error 2
make[1]: *** [Makefile:240: __sub-make] Error 2
make[1]: Leaving directory '/usr/lib/modules/6.9.6-arch1-1/build'
make: *** [Makefile:2: all] Error 2
```
Tried modifying Kbuild:
```bash
MODNAME     ?= nitara2
$(MODNAME)-m       += $(MODNAME).o module_hunter.o 
ccflags-y   += -Werror 
KBUILD_CFLAGS   := $(filter-out -pg,$(KBUILD_CFLAGS))
KBUILD_CFLAGS   := $(filter-out -mfentry,$(KBUILD_CFLAGS))
```
And `make` only created modules.order and Module.symvers

--- a/README.md
+++ b/README.md
@@ -13,4 +13,4 @@ As the original module was written at the time of 2.2—2.4 kernels and IA32 arc
 * module descriptor structs seem to get allocated within module mapping area and not vmalloc region, as was in the original work (see [Memory layout of x86-64](https://www.kernel.org/doc/Documentation/x86/x86_64/mm.txt))
 * module mapping area on x64 is about 12 times larger than vmalloc area on i386 (1520 MB vs 128M, regardless of paging levels), so more checks were added eliminate false positives that become more possible with a larger region, still without introducing false negatives.
 
-Tested on x86_64 4.4 (Ubuntu 16.04), 5.15 (Ubuntu 22.04), 5.19, 6.5 kernels
+Tested on x86_64 4.4 (Ubuntu 16.04), 5.15 (Ubuntu 22.04), 5.19, 6.5, 6.12 kernels

--- a/nitara2.c
+++ b/nitara2.c
@@ -163,7 +163,7 @@ static bool check_name_valid(char *s)
 #endif
 
 
-ssize_t showmodule_read(struct file *unused_file, char *buffer, size_t len, loff_t *off)
+static ssize_t showmodule_read(struct file *unused_file, char *buffer, size_t len, loff_t *off)
 {
     struct module *p;
     unsigned long i;

